src/libraries/System.Memory.Data/src/System/BinaryData.cs

```diff
@@ -48,10 +48,13 @@ public BinaryData(byte[] data)
         /// <param name="options">The options to use when serializing to JSON.</param>
         /// <param name="type">The type to use when serializing the data. If not specified, <see cref="object.GetType"/> will
         /// be used to determine the type.</param>
-#pragma warning disable AZC0014 // Avoid using banned types in public API
         public BinaryData(object? jsonSerializable, JsonSerializerOptions? options = default, Type? type = default)
-#pragma warning restore AZC0014 // Avoid using banned types in public API
         {
+            if (type == null)
+            {
+                type = jsonSerializable == null ? typeof(object) : jsonSerializable.GetType();
```

Comment:
This change was needed for two reasons.
1. passing null, null now throws in 5.0, so instead pass typeof(object).  https://docs.microsoft.com/en-us/dotnet/core/compatibility/serialization#jsonserializerserialize-throws-argumentnullexception-when-type-parameter-is-null
2. passing object, null never worked and this API is documented as calling `object.GetType()` so I did that.

Reply:
woops, not sure what happened there - I had this null checking in the prior version. Yes I think we will want to do a patch release for this.

Reply:
Ported here: https://github.com/Azure/azure-sdk-for-net/pull/16957